testsuite/MDAnalysisTests/topology/test_tprparser.py

```diff
@@ -43,115 +45,39 @@ class TPRAttrs(ParserBase):
 
     def test_moltypes(self, top):
         moltypes = top.moltypes.values
-        assert_array_equal(moltypes, self.ref_moltypes)
-
-
+        assert_equal(moltypes, self.ref_moltypes)
 
 
 class TestTPR(TPRAttrs):
     """
     this test the data/adk_oplsaa.tpr which is of tpx version 58
     """
-
-    __test__ = True
-
-    filename = TPR
     expected_n_atoms = 47681
     expected_n_residues = 11302
     expected_n_segments = 3
     ref_moltypes = np.array(['AKeco'] * 214 + ['SOL'] * 11084 + ['NA+'] * 4,
                             dtype=object)
 
+    @pytest.fixture()
+    def filename(self):
+        return TPR
+
 
 # The follow test the same system grompped by different version of gromacs
 # FORMAT: TPRABC, where numbers ABC indicates the version of gromacs that
 # generates the corresponding tpr file
-
-class TPRBase(TPRAttrs):
+class TestTPRBase(TPRAttrs):
     expected_n_atoms = 2263
     expected_n_residues = 230
     expected_n_segments = 2
     ref_moltypes = np.array(['Protein_A'] * 129 + ['SOL'] * 101, dtype=object)
 
+    @pytest.fixture(params=[TPR400, TPR402, TPR403, TPR404, TPR405, TPR406,
+                            TPR407, TPR450, TPR451, TPR452, TPR453, TPR454,
+                            TPR455, TPR502, TPR504, TPR505, TPR510, TPR2016])
+    def filename(self, request):
+        return request.param
```

Comment:
@jbarnoud This what you had in mind to remove the classes?

Reply:
@richardjgowers you are also OK with this?

Reply:
I'll have a play later today